src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -543,12 +543,12 @@ private unsafe int HandleEventShutdownComplete(ref SHUTDOWN_COMPLETE data)
                 (shutdownByApp: true, closedRemotely: true) => ThrowHelper.GetConnectionAbortedException((long)data.ConnectionErrorCode),
                 // It's local shutdown by app, this side called QuicConnection.CloseAsync, throw QuicError.OperationAborted.
                 (shutdownByApp: true, closedRemotely: false) => ThrowHelper.GetOperationAbortedException(),
-                // It's remote shutdown by transport, (TODO: we should propagate transport error code), throw QuicError.InternalError.
+                // TODO: we should propagate transport error code
```

Comment:
What is left to do for properly propagating transport error code? Isn't this already doing that?

Reply:
A code passed to GetExceptionForMsQuicStatus seems to be already passed as HResult https://github.com/dotnet/runtime/blob/51584ceecbcc38352676bf4b3b45b272e97cc99a/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ThrowHelper.cs#L55-L59

Reply:
Or it's not enough?

Reply:
That propagates the MsQuic status code, the comment (and related issue) is about the QUIC Transport Error sent over the wire.

Reply:
>  is about the QUIC Transport Error sent over the wire.

Yep, these ones https://www.rfc-editor.org/rfc/rfc9000.html#name-transport-error-codes. I think the work on msquic side is already done, we just don't use it yet since it requires new API.